mithril-signer/src/runtime/runner.rs

```diff
@@ -262,77 +255,53 @@ impl Runner for SignerRunner {
         Ok(())
     }
 
-    async fn can_i_sign(&self, pending_certificate: &CertificatePending) -> StdResult<bool> {
-        debug!("RUNNER: can_i_sign");
-        if self
+    async fn can_sign_current_epoch(&self) -> StdResult<bool> {
+        let epoch = self
```

Comment:
We can create a function to get the epoch_service in read mode
```
    async fn epoch_service_read(&self) -> RwLockReadGuard<'_, dyn EpochService> {
        self.services.epoch_service.read().await
    }
```

Then, code that use this epoch_service can be written in a shorter way
 ```
let epoch = self.epoch_service_read().await.epoch_of_current_data()?;
``

Reply:
Done ✅